### ISSUE TITLE
Search API command doesn't give indication of success

### ISSUE THREAD

**Issue by calebmeyer** [t9md/atom-vim-mode-plus#1068]
<details>
  <summary>debug info</summary>

```json
{
  "atom": "1.26.1",
  "platform": "win32",
  "release": "10.0.16299",
  "vmpVersion": "1.32.0",
  "vmpConfig": {
    "automaticallyEscapeInsertModeOnActivePaneItemChange": true,
    "keymapBackslashToInnerCommentOrParagraphWhenToggleLineCommentsIsPending": true,
    "keymapCCToChangeInnerSmartWord": true,
    "keymapPToPutWithAutoIndent": true,
    "keymapUnderscoreToReplaceWithRegister": true,
    "keymapYToYankToLastCharacterOfLine": true,
    "showHoverSearchCounter": true,
    "statusBarModeStringStyle": "long",
    "useSmartcaseForSearch": true,
    "useSmartcaseForSearchCurrentWord": true,
    "wrapLeftRightMotion": true,
    "blackholeRegisteredOperators": [],
    "startInInsertModeScopes": [],
    "charactersToAddSpaceOnSurround": [],
    "highlightSearchExcludeScopes": [],
    "flashOnOperateBlacklist": []
  }
}
```

</details>

Read and check all "Check list" bellow.

# Check list

You have to check all before open issue.

- [x] Provide your environment info clipped by `Vim Mode Plus: Clip Debug Info` command.
- [x] Try with latest Atom and latest vim-mode-plus.
- [x] Pick descriptive and non-ambiguous subject
- [x] Express "what" you want(feature? config option?, behavior change?) in short sentence(not long!).
- [x] Contrast current behavior if you want to change current behavior, with sample text, operation(keystroke) and result.
- [x] Include real use case so that maintainer can understand "why" you need help.
- [x] Include Atom(`atom --version`), vim-mode-plus version, and OS version(e.g. macOS Sierra 10.12.3).
- [x] If keybinding issue, Read [this](https://github.com/t9md/atom-vim-mode-plus/wiki/IssueReport#some-keybinding-not-working).

## What
I would like for the search commands to globally store or return some programmatic indication whether the search found any results.

I am aware that searching gives auditory (beep) feedback and visual (flash) feedback, but I don't know that either of those have happened when I dispatch a search command.

## Why
I'm maintaining a simplified macro system for vim mode plus over at https://github.com/calebmeyer/vim-mode-plus-macros . This is one of the pieces I think I need to properly support https://github.com/calebmeyer/vim-mode-plus-macros/issues/4 . If I know whether a search is failed, I can stop processing a macro once I hit a search failure.

## Versions
```
$ apm --version
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.26.1
python 3.6.5
git 2.16.2.windows.1
visual studio

vim-mode-plus 1.32.0

Windows 10 Fall Creators' Update (10.0.16299)
```

**Comment by calebmeyer** [comment]
The code I'm using to test this is on my fix-search branch on the vim-mode-plus-macros repo. When I run a macro that includes a search, the search commands give me a promise, which is resolved as `[undefined]` (an array containing one element, `undefined`).

[Here's the console log statement](https://github.com/calebmeyer/vim-mode-plus-macros/blob/6e1eeabc1dee8f9d2efa254921d5b514ddda5f58/lib/vim-mode-plus-macros.js#L66)

Here's a successful search (found the `h` character):
![image](https://user-images.githubusercontent.com/3342777/40024028-916afec2-5792-11e8-8914-e9e7ae8c352c.png)

And here's a failed search (did not find `z`):
![image](https://user-images.githubusercontent.com/3342777/40024051-abe942cc-5792-11e8-9c8b-576835836d53.png)

**Comment by t9md** [comment]
Sorry for my late response.
Before going to implement your request, I want clarify basic understanding.

Could you replay macro which include search(`/`) as long as search can find match?
I didn't check detail of your packages, but implementing macro on Atom editor is not easy in some situation.
When you search(`/`) keyword normally, activeElement become search's mini-editor then your keystrokes would be sent to that activeElement.
When you replay keystroke by dispatch, you need to keep track of shift of activeElement which might or might not be introduced by each keystroke.

Btw what is "Search API command"  you are mentioning?

**Comment by calebmeyer** [comment]
Search API command is the result of dispatching `core:confirm` against the mini-editor. I believe it runs [search](https://github.com/t9md/atom-vim-mode-plus/blob/master/lib/search-model.js#L123) internally.

**Comment by t9md** [comment]
Ok, its not search API, it’s just atom’s api yo dispatch command programatically.

So, to simplify the issue are you saying you want the promise retuned by core:confirm to the search-mini-editor return different value depending of result, which is curtently same undefined valued arrray.

Am I understanding your request correctly?

**Comment by calebmeyer** [comment]
Yes. I would either like to be able to determine the success/failure of the search as the result of dispatching core:confirm, or I would like another way to determine it. 

Vim-mode-plus knows whether the last search command succeeded or failed, since it has search-state and it is giving me a beep/flash. I just want a way to find that out from my package.
